drawers/maxicode.go

```diff
@@ -0,0 +1,74 @@
+package drawers
+
+import (
+	"bytes"
+	"fmt"
+	"image"
+	"strings"
+
+	"github.com/fogleman/gg"
+	"github.com/ingridhq/maxicode"
+	"github.com/ingridhq/zebrash/elements"
+)
+
+func NewMaxicodeDrawer() *ElementDrawer {
+	return &ElementDrawer{
+		Draw: func(gCtx *gg.Context, element interface{}, options DrawerOptions) error {
+			barcode, ok := element.(*elements.MaxicodeWithData)
+			if !ok {
+				return nil
+			}
+
+			const RS = maxicode.RS
+			const GS = maxicode.GS
+			const codeHeader = "[)>" + RS + "01" + GS
+
+			headerPos := strings.Index(barcode.Data, codeHeader)
+
+			if headerPos < 0 || len(barcode.Data) < headerPos+9 {
+				return fmt.Errorf("invalid length of maxicode data")
+			}
+
+			mainData := barcode.Data[:headerPos]
+			addData := barcode.Data[headerPos:]
+			headerData := addData[0:9]
```

Comment:
Move 9 to a constant, it appears also above

Reply:
Good advice, moved